src/understand-tidb/plan-cache.md

```diff
@@ -1 +1,59 @@
 # Plan Cache
+
+TiDB supports execution plan caching for `Prepare` and `Execute` queries. By using cached plan, TiDB can skip the optimization phase and then gain some performance benefits.
+
+The flow of handling `Prepare` and `Execution` is shown below:
+
+![plan-cache-overview](../img/plan-cache-overview.png)
+
+There still are some limitations about TiDB plan cache, and please see [the TiDB Plan Cache Doc](https://docs.pingcap.com/tidb/stable/sql-prepare-plan-cache) for more details.
+
+# The flow of handling Prepare statement 
+
+The flow of handling `Prepare` statement is not complex, and the main work are below:
+1. parse all parameter placeholders, 
+2. parse the query into an AST,
+3. store them into current session's `PrepareStmtCache`.
+
+The entry function is `clientConn.handleStmtPrepare`, and most work is done in `PrepareExec.Next`.
```

Comment:
For the cache part, I found other logic in

* common_plans.go: func (e *Execute) getPhysicalPlan(ctx context.Context, sctx sessionctx.Context, is infoschema.InfoSchema, preparedStmt *CachedPrepareStmt) error

where `sctx.PreparedPlanCache().Put(cacheKey, cacheVals)` etc. called.